utilities/io500_fixed.sh

```diff
@@ -5,29 +5,71 @@
 
 set -euo pipefail   # give bash better error handling.
 export LC_NUMERIC=C  # prevents printf errors
+export LC_ALL=C
+IO500_MIN_WRITE_RUNTIME=300
 
 function main {
+  set_defaults
   check_variables
   output_description
   core_setup
+  clean_cache
   ior_easy "write"
+  clean_cache
   mdt_easy "write"
+  clean_cache
   touch $timestamp_file  # this file is used subsequently by the find command
   ior_hard "write"
+  clean_cache
   mdt_hard "write"
+  clean_cache
   myfind
   ior_easy "read"
+  clean_cache
   mdt_easy "stat"
+  clean_cache
   ior_hard "read"
+  clean_cache
   mdt_hard "stat"
+  clean_cache
   mdt_easy "delete"
+  clean_cache
   mdt_hard "read"
+  clean_cache
   mdt_hard "delete"
   mdreal
   cleanup
   output_score
 }
 
+function set_defaults {
+  # Set default values, they may be overwritten by the user if necessary
+  io500_mdtest_easy_files_per_proc=${io500_mdtest_easy_files_per_proc:-250000}
+  io500_mdtest_hard_files_per_proc=${io500_mdtest_hard_files_per_proc:-500000}
+  io500_ior_hard_writes_per_proc=${io500_ior_hard_writes_per_proc:-100000}
+  # io500_ior_easy_size is the amount of data written per rank in MiB units,
+  # but it can be any number as long as it is somehow used to scale the IOR
+  # runtime as part of io500_ior_easy_params
+  io500_ior_easy_size=${io500_ior_easy_size:-200000}
+  # 2M writes, 2 GB per proc, file per proc
+  io500_ior_easy_params=${io500_ior_easy_params:-"-t 2048k -b ${io500_ior_easy_size}m -F"}
+  io500_mdreal_params=${io500_mdreal_params:-"-P=5000 -I=1000"}
+  io500_clean_cache_cmd=${io500_clean_cache_cmd:-drop_cache}
+}
+
+function drop_cache {
+  sudo bash -c "echo 3 > /proc/sys/vm/drop_caches"
```

Comment:
What happens if this is run by regular user?  Won't this interrupt the script and ask for a password?

Reply:
Basically the user needs to add an option to allow mpirun execution as root also (--allow-run-as-root). This is though bad practice but not sure what is the best thing here.

Reply:
sudo(8) shows:

     -n, --non-interactive
                   Avoid prompting the user for input of any kind.  If a
                   password is required for the command to run, sudo will
                   display an error message and exit.

so use `sudo -n` and append `|| true` at the end so that the failure to clear the cache is not considered an error.